python/lsst/ts/externalscripts/auxtel/latiss_take_flats.py

```diff
@@ -0,0 +1,548 @@
+        self.simulation_mode = simulation_mode
+        # self.filename = None
+
+        self.image_in_oods_timeout = 15.0
+        self.get_image_timeout = 10.0
+
+    async def run(self, simulation_mode: int = 0):
+
+        await self.arun(checkpoint_active=True, simulation_mode=simulation_mode)
+
+    @classmethod
+    def get_schema(cls):
+        yaml_schema = """
+$schema: http://json-schema/draft-07/schema#
+$id: https://github.com/lsst-ts/ts_externalscripts/auxtel/LatissTakeFlats.yaml
+title: LatissTakeFlats v1
+description: Configuration for LatissTakeFlats.
+type: object
+properties:
+    latiss_filter:
+        type: string
+        default: 'empty_1'
+    latiss_grating:
+        type: string
+        default: 'empty_1'
+    sequence:
+        description: Yaml string with all required data.
+        anyOf:
+            - type: string
+            - type: "null"
+        default: null
+
+    s3instance:
+        type: string
+        default: 'cp'
+
+required: [latiss_filter, latiss_grating]
+additionalProperties: false
+            """
+        return yaml.safe_load(yaml_schema)
+
+    def set_metadata(self, metadata):
+
+        self.log.debug("In set_metadata")
+        metadata.nimages = 3
+        metadata.duration = 600  # set arbitrarily
+
+    async def configure(self, config):
+        """Configure script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+        self.log.debug(f"Configuration: {config}")
+
+        self.config = config
+        self.s3instance = config.s3instance
+
+        self.log.debug(f"in config, {self.sequence=}")
+
+        if config.sequence is not None:
+            self.log.debug("Loading custom configuration")
+            self.sequence = yaml.safe_load(config.sequence)
+
+    async def handle_checkpoint(self, checkpoint_active, checkpoint_message):
+
+        if checkpoint_active:
+            await self.checkpoint(checkpoint_message)
+        else:
+            self.log.info(checkpoint_message)
+
+    def get_flat_sequence(self, latiss_filter: str, latiss_grating: str):
+        """Return the pre-determined flat field sequences for a given
+        LATISS filter and grating.
+
+               Parameters:
+               -----------
+
+               filt : `string`
+                   LATISS filter.
+
+               grating : `string`
+                   LATISS grating.
+
+               Returns
+               -------
+
+               sequence : `list`
+                   A list of sequences (dicts) for a given LATISS filter
+                  and grating combination.
+
+        """
+        self.log.debug(
+            f"Running get_flat_sequence with the default for {latiss_filter=} and {latiss_grating=}"
+        )
+
+        # Check that the LATISS filter/grating combination is valid
+        if latiss_filter == "SDSSr_65mm" and latiss_grating == "empty_1":
```

Comment:
Yes! That's the plan!
We need to establish configs for each filter+grating combination. 
Question for bonus points. Where do you think those sequence files belong?

Reply:
Good question. I see that a default sequence for the test cases lives inside the tests/data/auxtel. I can also see that there is a data/scripts folder for the executable scripts, but I don't see a similar data folder inside the main scripts folder (/externalscripts/auxtel). I was looking around (e.g. ts_standardscripts) and could not find a similar case, so not much in favor of creating a data folder inside the main scripts folder. Maybe it could live in the lsst-ts/ts_config_ocs, where the scheduler/survey config files live.